platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -1025,9 +1027,25 @@ static void schedule_order_unlock(unsigned lock_index)
 	atomic_store_release(&rctx->rwin->olock[lock_index],
 			     rctx->sn + 1,
 			     /*readonly=*/false);
+	atomic_store_release(&rctx->rwin->lock_acquired,
+			     rctx->rwin->lock_count + 1, false);
 	rctx->olock_flags |= 1U << lock_index;
 }
 
+static void schedule_order_unlock_lock(unsigned lock_index)
+{
+	struct reorder_context *rctx;
+
+	rctx = sched_ts->rctx;
+	if (odp_unlikely(rctx == NULL || rctx->rwin == NULL)) {
+		ODP_ERR("Invalid call to odp_schedule_order_unlock_lock\n");
+		return;
+	}
```

Comment:
This can be optimized into an `ODP_ASSERT()` to be consistent with the other schedulers. Behavior is undefined if these routines are not called within an ordered context, so no need to take a runtime hit to do these type of checks.